### PR TITLE
Feature/sharing settings

### PR DESCRIPTION
**Pull request by Hamik27**
https://ekoapp.atlassian.net/browse/UP-708

Share post event handler in UIKit sdk, implementation in sample app.

### CODE REVIEW

File: UpstraUIKit/SampleApp/SampleApp/AppDelegate.swift [hunk 1]
```diff
@@ -88,4 +88,10 @@ class CustomEventHandler: EkoEventHandler {
         }
     }
     
+    override func sharePostDidTap(from source: EkoViewController, postId: String) {
+        let urlString = "https://amity.co/posts/\(postId)"
+        guard let url = NSURL(string: urlString) else { return }
```
Comment (sainttail) [inline UpstraUIKit/SampleApp/SampleApp/AppDelegate.swift]: I think we can just use `URL(string:)` no need to use `NSURL` from obj-c counterpart

**Review by sainttail — APPROVED** [review]

File: UpstraUIKit/SampleApp/SampleApp/AppDelegate.swift (hunk 1 above)
Comment (Hamik27) [reply UpstraUIKit/SampleApp/SampleApp/AppDelegate.swift]: Done

File: UpstraUIKit/UpstraUIKit/EkoEventHandler.swift [hunk 2]
```diff
@@ -109,4 +109,12 @@ open class EkoEventHandler {
         source.present(navigationController, animated: true, completion: nil)
     }
     
+    /// Event to share a post
+    /// It will be triggered when share button is tapped
```
Comment (nrlnishan) [inline UpstraUIKit/UpstraUIKit/EkoEventHandler.swift]: I think the comment is a bit misleading. When a share button is tapped, we open the bottom sheet and this event gets triggered only when more options is tapped. Please revise this comment

**Review by nrlnishan — APPROVED** [review]

File: UpstraUIKit/UpstraUIKit/EkoEventHandler.swift (hunk 2 above)
Comment (Hamik27) [reply UpstraUIKit/UpstraUIKit/EkoEventHandler.swift]: Done

**Pull request merged**